safenode/src/network/mod.rs

```diff
@@ -145,26 +144,31 @@ impl SwarmDriver {
             NonZeroUsize::new(CLOSE_GROUP_SIZE).ok_or_else(|| Error::InvalidCloseGroupSize)?,
         );
 
-        let request_response = request_response::Behaviour::new(
-            MsgCodec(),
-            iter::once((MsgProtocol(), ProtocolSupport::Outbound)),
-            Default::default(),
-        );
-
-        Self::with(cfg, request_response)
+        Self::with(cfg, ProtocolSupport::Outbound)
     }
 
     // Private helper to create the network components with the provided config and req/res behaviour
     fn with(
-        cfg: KademliaConfig,
-        request_response: request_response::Behaviour<MsgCodec>,
+        kad_cfg: KademliaConfig,
+        req_res_protocol: ProtocolSupport,
     ) -> Result<(Network, mpsc::Receiver<NetworkEvent>, SwarmDriver)> {
         // Create a random key for ourself.
         let keypair = identity::Keypair::generate_ed25519();
         let peer_id = PeerId::from(keypair.public());
 
         info!("Node (PID: {}) with PeerId: {peer_id}", std::process::id());
 
+        // RequestResponse configuration
+        let mut req_res_config = RequestResponseConfig::default();
+        let _ = req_res_config.set_request_timeout(REQUEST_TIMEOUT);
+        let _ = req_res_config.set_connection_keep_alive(CONNECTION_KEEP_ALIVE_TIMEOUT);
+
+        let request_response = request_response::Behaviour::new(
```

Comment:
Shouldn't this variable be called `behaviour`?
`request_response` is a very fuzzy name. It is a Behaviour instance, right?

Reply:
I've renamed it to `req_res_behaviour` as we have multiple structs implementing the `Behaviour` trait.

Reply:
I think we have a couple of other structs that implement the `Behaviour` trait, and they're just called by their crate/struct name, e.g. `kademlia`, `mdns`. So I think `request_response` fits well in this context? 

The variable names are self-explanatory inside this function, but adding `_behaviour` to them is also an option.

Reply:
Yeah, will mull this over, later.. :)